pkg/controller/prometheus.go

```diff
@@ -224,27 +226,20 @@ func (c *Controller) deletePrometheusAddlConfig(ctx context.Context, tenant *min
 		return err
 	}
 
-	var scrapeConfigs []configmaps.ScrapeConfig
+	var scrapeConfigs, exceptedScrapeConfigs []configmaps.ScrapeConfig
 	err = yaml.Unmarshal(secret.Data[miniov2.PrometheusAddlScrapeConfigKey], &scrapeConfigs)
 	if err != nil {
 		return err
 	}
-	// Check if the scrape config is present
-	hasScrapeConfig := false
-	scIndex := -1
-	for i, sc := range scrapeConfigs {
-		if sc.JobName == tenant.PrometheusOperatorAddlConfigJobName() {
-			hasScrapeConfig = true
-			scIndex = i
-			break
+	for _, sc := range scrapeConfigs {
+		if !strings.HasPrefix(sc.JobName, tenant.PrometheusOperatorAddlConfigJobName()) {
+			exceptedScrapeConfigs = append(exceptedScrapeConfigs, sc)
 		}
 	}
-	if hasScrapeConfig {
+	if !reflect.DeepEqual(scrapeConfigs, exceptedScrapeConfigs) {
 		klog.Infof("Deleting MinIO tenant Prometheus scrape config")
-		// Delete the config
-		newScrapeConfigs := append(scrapeConfigs[:scIndex], scrapeConfigs[scIndex+1:]...)
 		// Update the secret
-		scrapeCfgYaml, err := yaml.Marshal(newScrapeConfigs)
+		scrapeCfgYaml, err := yaml.Marshal(exceptedScrapeConfigs)
```

Comment:
`reflectDeepEqual` will always return false due to the implementation of `GetPrometheusConfig`. This will trigger multiple unnecessary secret recreation.

Please consider the following option instead:
```
// add new scrape configs
    exceptedScrapeConfigs = append(exceptedScrapeConfigs, promCfg.ScrapeConfigs...)
    // compare scrape configs excluding bearer token to avoid continuous updates
    updateScrapeConfig := false
    if len(scrapeConfigs) != len(exceptedScrapeConfigs) {
      updateScrapeConfig = true
    } else {
      for i, _ := range scrapeConfigs {
        if scrapeConfigs[i].JobName != exceptedScrapeConfigs[i].JobName ||
          scrapeConfigs[i].MetricsPath != exceptedScrapeConfigs[i].MetricsPath ||
          scrapeConfigs[i].Scheme != exceptedScrapeConfigs[i].Scheme ||
          !reflect.DeepEqual(scrapeConfigs[i].TLSConfig, exceptedScrapeConfigs[i].TLSConfig) ||
          !reflect.DeepEqual(scrapeConfigs[i].StaticConfigs, exceptedScrapeConfigs[i].StaticConfigs) {
          updateScrapeConfig = true
          break
        }
      }
    }
    if updateScrapeConfig {
      klog.Infof("Adding MinIO tenant Prometheus scrape config")
      scrapeConfigs = exceptedScrapeConfigs
      scrapeCfgYaml, err := yaml.Marshal(scrapeConfigs)
      if err != nil {
        return err
      }
      secret.Data[miniov2.PrometheusAddlScrapeConfigKey] = scrapeCfgYaml
      _, err = c.kubeClientSet.CoreV1().Secrets(ns).Update(ctx, secret, metav1.UpdateOptions{})
      if err != nil {
        return err
      }
    }
```

Reply:
please take a look @jiuker